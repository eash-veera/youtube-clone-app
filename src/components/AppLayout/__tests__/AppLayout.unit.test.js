import React from 'react';
import { shallow } from 'enzyme';
import { AppLayout } from '../AppLayout';

describe('AppLayout', () => {
    
    test('renders empty <AppLayout />', () => {
        const wrapper = shallow(
            <AppLayout />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders <AppLayout /> with one child', () => {
        const wrapper = shallow(
            <AppLayout>
                <div>Child 1</div>
            </AppLayout>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders <AppLayout /> with children', () => {
        const wrapper = shallow(
            <AppLayout>
                <div>Child 1</div>
                <div>
                    <div>Child 2</div>
                    <span>Child 3</span>
                </div>
            </AppLayout>
        );
        expect(wrapper).toMatchSnapshot();
    });
});