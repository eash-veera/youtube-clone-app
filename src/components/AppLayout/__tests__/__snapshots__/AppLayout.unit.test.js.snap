// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout renders <AppLayout /> with children 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
  <div>
    Child 1
  </div>
  <div>
    <div>
      Child 2
    </div>
    <span>
      Child 3
    </span>
  </div>
</div>
`;

exports[`AppLayout renders <AppLayout /> with one child 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
  <div>
    Child 1
  </div>
</div>
`;

exports[`AppLayout renders empty <AppLayout /> 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
</div>
`;
